src/main/kotlin/com/openlattice/users/Auth0Utils.kt

```diff
@@ -6,20 +6,81 @@ import com.auth0.exception.Auth0Exception
 import com.auth0.json.mgmt.users.User
 import com.auth0.json.mgmt.users.UsersPage
 import com.auth0.jwt.algorithms.Algorithm
+import com.dataloom.mappers.ObjectMappers
 import com.geekbeast.auth0.*
 import com.openlattice.authentication.Auth0AuthenticationConfiguration
+import com.openlattice.users.export.*
+import org.slf4j.LoggerFactory
+import java.io.BufferedReader
+import java.io.InputStreamReader
 import java.security.InvalidAlgorithmParameterException
+import java.time.Instant
+import java.util.stream.Collectors
+import java.util.zip.GZIPInputStream
 
 /**
  *
  * @author Matthew Tamayo-Rios &lt;matthew@openlattice.com&gt;
  */
+
+private val logger = LoggerFactory.getLogger("Auth0Utils")
+
+/**
+ * Calls an export job to download all users from auth0 to a json and parses it to a sequence of users.
+ *
+ * Note: This export is used, because the user API has a 1000 user limitation.
+ * @see <a href="https://auth0.com/docs/users/search/v3/get-users-endpoint#limitations"> Auth0 user endpoint
+ * limitations </a>
+ */
+fun getUsers(auth0ApiExtension: Auth0ApiExtension): List<User> {
+    val exportEntity = auth0ApiExtension.userExport()
+    val job = exportEntity.submitExportJob(
+            UserExportJobRequest(listOf(USER_ID, EMAIL, NICKNAME, APP_METADATA, IDENTITIES))
+    )
+
+    // will fail if export job hangs too long with too many requests error (429 status code)
+    // https://auth0.com/docs/policies/rate-limits
+    var exportJobResult = exportEntity.getJob(job.id)
+    while (exportJobResult.status == JobStatus.pending) {
+        exportJobResult = exportEntity.getJob(job.id)
```

Comment:
This needs some kind of time-based backoff -- hammering the status endpoint is likely to get us throttled/blacklisted.